pkg/apis/pipeline/v1beta1/pipeline_types_test.go

```diff
@@ -1057,3 +1074,49 @@ func TestPipelineTask_ValidateEmbeddedOrType(t *testing.T) {
 		})
 	}
 }
+
+func TestPipelineTask_IsMatrixed(t *testing.T) {
+	type arg struct {
+		*Matrix
+	}
+	testCases := []struct {
+		name     string
+		arg      arg
+		expected bool
+	}{
+		{
+			name: "nil matrix",
+			arg: arg{
+				Matrix: nil,
+			},
+			expected: false,
+		},
+		{
+			name: "empty matrix",
+			arg: arg{
+				Matrix: &Matrix{},
+			},
+			expected: false,
+		},
+		{
+			name: "matrixed",
+			arg: arg{
+				Matrix: &Matrix{
+					Params: []Param{{Name: "platform", Value: ParamValue{ArrayVal: []string{"linux", "windows"}}}},
+				},
+			},
+			expected: true,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			pt := &PipelineTask{
+				Matrix: tc.arg.Matrix,
+			}
+			isMatrixed := pt.IsMatrixed()
+			if isMatrixed != tc.expected {
+				t.Errorf("PipelineTask.IsMatrixed() return bool: %v, but wanted: %v", isMatrixed, tc.expected)
+			}
+		})
+	}
+}
```

Comment:
this test needs to be added in v1 too

Reply:
thanks @jerop , i've added this test to v1